### PR TITLE
0.1.1 19 local_data_platform.source.near.bigquery.get(query)

### PR DESCRIPTION
**Pull request by mrutunjay-kinagi**

### CODE REVIEW

File: local-data-platform/local_data_platform/source/gcp/gcp_bigquery/source_gcp_bigquery.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+from source.gcp.gcp_connection import GCPBigQueryConnection
+
+class SourceGCPBigQuery:
+    """
+    A class to interact with Google BigQuery.
+    client : GCPBigQueryConnection
+        An instance of GCPBigQueryConnection to interact with BigQuery.
+    """
+    def __init__(self):
+        self.client = GCPBigQueryConnection('path/to/credentials.json', 'your-project-id')
```
Comment (tusharchou) [inline local-data-platform/local_data_platform/source/gcp/gcp_bigquery/source_gcp_bigquery.py:10]: This looks great! 👍

File: local-data-platform/local_data_platform/source/gcp/gcp_bigquery/source_gcp_bigquery.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+    def fetch_data(self, query):
+        """
+        Executes a given SQL query on Google BigQuery and returns the result as a pandas DataFrame.
+
+        Args:
+            query (str): The SQL query to be executed.
+
+        Returns:
+            pandas.DataFrame: The result of the query as a DataFrame.
```
Comment (tusharchou) [inline local-data-platform/local_data_platform/source/gcp/gcp_bigquery/source_gcp_bigquery.py:20]: Really detailed documentation

File: local-data-platform/local_data_platform/source/gcp/gcp_connection.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        for the project specified by `self.project_id`.
+
+        Returns:
+            bigquery.Client: An authenticated BigQuery client.
+        """
+        credentials = service_account.Credentials.from_service_account_file(self.credentials_path)
+        client = bigquery.Client(credentials=credentials, project=self.project_id)
+        return client
+
+    def query(self, query_string):
```
Comment (tusharchou) [inline local-data-platform/local_data_platform/source/gcp/gcp_connection.py:48]: maybe this is could be moved to big query class

File: local-data-platform/local_data_platform/source/near/source_near_gcp.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+Classes:
+    Source_Near_GCP: A class to fetch NEAR protocol transaction data from GCP BigQuery.
+
+Methods:
+    fetch_near_transaction(self):
+        Fetches NEAR protocol transaction data from the previous day.
+        Returns:
+            DataFrame: A pandas DataFrame containing the transaction data.
+"""
+class Source_Near_GCP(SourceGCPBigQuery):
```
Comment (tusharchou) [inline local-data-platform/local_data_platform/source/near/source_near_gcp.py:15]: this class should have columns required we can come up with a list together

**Review by tusharchou — APPROVED** [review]
All these comments can be addressed later LGTM 👍

**Comment by mrutunjay-kinagi** [comment]
Added pytest  files.

File: local-data-platform/local_data_platform/source/gcp/gcp_connection.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        
+        query(query_string):
+            Executes a SQL query on BigQuery and returns the results.
+
+
+        Example usage:
+        connection = GCPBigQueryConnection('/path/to/credentials.json', 'your-project-id')
+        results = connection.query('SELECT * FROM your_dataset.your_table')
+        for row in results:
+            print(row)
```
Comment (tusharchou) [inline local-data-platform/local_data_platform/source/gcp/gcp_connection.py:26]: Exceptional documentation!!!

File: local-data-platform/local_data_platform/source/near/source_near_gcp.py (hunk 4 above)
Comment (tusharchou) [inline local-data-platform/local_data_platform/source/near/source_near_gcp.py:15]: Class name can be camel case

File: local-data-platform/local_data_platform/source/near/source_near_gcp.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+    def fetch_near_transaction(self):
+        query = """
+            SELECT *
+            FROM `bigquery-public-data.crypto_near.transactions`
```
Comment (tusharchou) [inline local-data-platform/local_data_platform/source/near/source_near_gcp.py:20]: Can we read the sql from a file will make it configurable

File: local-data-platform/tests/test_gcp_connection.py [hunk 7]
```diff
@@ -0,0 +1,88 @@
+import pytest
```
Comment (tusharchou) [inline local-data-platform/tests/test_gcp_connection.py:1]: This is what makes a good lib great!

**Review by tusharchou — CHANGES_REQUESTED** [review]

**Review by tusharchou — APPROVED** [review]
Lets test this with a demo

**Comment by redpheonixx** [comment]
> Added pytest files.

### pytests are failing
image

The error message indicates that Python is unable to import the `GCPBigQueryConnection` from the specified module. Here are some potential solutions:

1. **Check the Import Path**: Ensure that the path to `GCPBigQueryConnection` is correct and that it exists in `local_data_platform/store/source/gcp/bigquery/__init__.py`.
   
2. **Correct Package and Module Names**: Make sure that all package and module names are correctly spelled and follow Python naming conventions.

3. **Ensure `__init__.py` Exists**: Verify that an `__init__.py` file is present in the `local_data_platform`, `store`, `source`, `gcp`, and `bigquery` directories to make them proper Python packages.

4. **Relative Import Issues**: If you are using relative imports, ensure they are correctly specified. Sometimes converting relative imports to absolute imports can solve the issue.

5. **Check for Circular Imports**: Make sure that your code does not have circular imports, where two or more modules are trying to import each other.

Here's an example of what the `__init__.py` might look like:

```python
# local_data_platform/store/source/gcp/bigquery/__init__.py

from .bigquery import GCPBigQueryConnection
```

If all these steps are correctly followed and the issue still persists, please share more details or additional error messages for further troubleshooting. Good luck! 🚀

**Pull request merged**